db/skipped_sequence.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2016-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+const (
+	ClipCapacityHeadroom = 100
+)
+
+type SkippedSequenceListInterface interface {
```

Comment:
```suggestion
type SkippedSequenceListEntryInterface interface {
```
The previous name gave the impression it was an interface for the list itself, not an entry on the list.  I'm on the fence about including 'Interface' in the name of an interface, but it's not unreasonable.